src/package.ts

```diff
@@ -30,28 +32,147 @@ export interface Package {
+  packageDirectoryPath,
+  projectDirectoryPath,
+  projectTagNames,
+}: {
+  packageDirectoryPath: string;
+  projectDirectoryPath: string;
+  projectTagNames: string[];
+}): Promise<Package> {
   const manifestPath = path.join(packageDirectoryPath, MANIFEST_FILE_NAME);
   const changelogPath = path.join(packageDirectoryPath, CHANGELOG_FILE_NAME);
   const { unvalidated: unvalidatedManifest, validated: validatedManifest } =
     await readPackageManifest(manifestPath);
+  const expectedReleaseTagNames = generateMonorepoRootPackageReleaseTagNames(
+    validatedManifest.version.toString(),
+  );
+  const tagNameMatchingLatestRelease = expectedReleaseTagNames.find(
+    (tagName) => {
+      return projectTagNames.includes(tagName);
+    },
+  );
+  // TODO: What if tags exist (projectTagNames.length > 0) but there is no tag
+  // for the current release (tagNameMatchingLatestRelease === undefined)?
+  // Should we fall back to the most recently created tag?
+  const hasChangesSinceLatestRelease =
+    tagNameMatchingLatestRelease === undefined
+      ? true
+      : await hasChangesInDirectorySinceGitTag(
+          projectDirectoryPath,
+          packageDirectoryPath,
+          tagNameMatchingLatestRelease,
+        );
+
+  return {
+    directoryPath: packageDirectoryPath,
+    manifestPath,
+    validatedManifest,
+    unvalidatedManifest,
+    changelogPath,
+    hasChangesSinceLatestRelease,
+  };
+}
+
+/**
+ * Collects information about a workspace package within a monorepo.
+ *
+ * @param args - The arguments to this function.
+ * @param args.packageDirectoryPath - The path to a package within a project.
+ * @param args.rootPackageVersion - The version of the root package of the
+ * monorepo to which this package belongs.
+ * @param args.projectDirectoryPath - The path to the project directory.
+ * @param args.projectTagNames - The tag names across the whole project.
+ * @returns Information about the package.
+ */
+export async function readMonorepoWorkspacePackage({
+  packageDirectoryPath,
+  rootPackageVersion,
+  projectDirectoryPath,
+  projectTagNames,
+}: {
+  packageDirectoryPath: string;
+  rootPackageVersion: SemVer;
+  projectDirectoryPath: string;
+  projectTagNames: string[];
+}): Promise<Package> {
+  const manifestPath = path.join(packageDirectoryPath, MANIFEST_FILE_NAME);
+  const changelogPath = path.join(packageDirectoryPath, CHANGELOG_FILE_NAME);
+  const { unvalidated: unvalidatedManifest, validated: validatedManifest } =
+    await readPackageManifest(manifestPath);
+  const expectedReleaseTagNames =
+    generateMonorepoWorkspacePackageReleaseTagNames(
+      validatedManifest.name,
+      validatedManifest.version.toString(),
+      rootPackageVersion.toString(),
+    );
+  const tagNameMatchingLatestRelease = expectedReleaseTagNames.find(
+    (tagName) => {
+      return projectTagNames.includes(tagName);
+    },
+  );
+  // TODO: What if tags exist (projectTagNames.length > 0) but there is no tag
```

Comment:
@Gudahtt What do you think would be a good way to handle this?

Reply:
I think the answer in [my other comment](https://github.com/MetaMask/create-release-branch/pull/23#discussion_r978117112) applies here as well.